### PR TITLE
Check if both currencies are enabled on market enabling (closes #1242)

### LINKED ISSUE #1242: Check if both currencies are enabled on market enabling
We can still enable market with disabled currency
1. Create market with enabled currencies
2. Disable currency (market is disabled too)
3. Enable market

### PR DESCRIPTION
**Pull request by ysv**

### CODE REVIEW

**Comment by ysv** [comment]
@yivo review please

File: spec/factories/currencies.rb [hunk 1]
```diff
@@ -12,6 +12,16 @@
       withdraw_fee         0.1
     end
 
+    trait :eur do
+      code                 'eur'
+      symbol               '€'
+      type                 'fiat'
+      precision            2
```
Comment (yivo) [inline spec/factories/currencies.rb]: Use 8

File: spec/factories/currencies.rb [hunk 2]
```diff
@@ -12,6 +12,16 @@
...
+      quick_withdraw_limit 10
```
Comment (yivo) [inline spec/factories/currencies.rb]: Set 1000

File: app/models/market.rb [hunk 3]
```diff
@@ -31,6 +31,7 @@ class Market < ActiveRecord::Base
   validates :ask_precision, :bid_precision, :position, numericality: { greater_than_or_equal_to: 0, only_integer: true }
   validates :ask_unit, :bid_unit, inclusion: { in: -> (_) { Currency.codes } }
   validate  :precisions_must_be_same
+  validate  :units_must_be_enabled, if: -> (market){ market.enabled? }
```
Comment (yivo) [inline app/models/market.rb]: Change to `if: :enabled?`

File: app/models/market.rb [hunk 4]
```diff
@@ -97,6 +98,12 @@ def precisions_must_be_same
       errors.add(:bid_precision, :invalid)
     end
   end
+
+  def units_must_be_enabled
+    [bid_unit, ask_unit].each do |unit|
+      errors.add(:base, 'Currencies must be enabled.') if Currency.lock.find_by_id(unit)&.disabled?
```
Comment (yivo) [inline app/models/market.rb]: We need to do some updates to it.

The error should look like `errors.add(:ask_unit, 'is not enabled.')`

```
[:bid_unit, :ask_unit].each do |unit|
  errors.add(unit, 'is not enabled.') if Currency.lock.find_by_id(public_send(unit))&.disabled?
end
```

**Review by yivo — CHANGES_REQUESTED** [review]

**Review by yivo — APPROVED** [review]

**Pull request merged**